src/lib/AirspeedValidator/AirspeedValidator.hpp

```diff
@@ -0,0 +1,178 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file AirspeedValidator.hpp
+ * Calculates airspeed from differential pressure and checks if this airspeed is valid.
+ */
+
+#pragma once
+
+#include <airspeed/airspeed.h>
+#include <ecl/airdata/WindEstimator.hpp>
+#include <uORB/topics/wind_estimate.h>
+
+
+using matrix::Dcmf;
+using matrix::Quatf;
+using matrix::Vector2f;
+using matrix::Vector3f;
+
+using namespace time_literals;
+
+struct airspeed_validator_update_data {
+	uint64_t timestamp;
+	float airspeed_indicated_raw;
+	float airspeed_true_raw;
+	uint64_t airspeed_timestamp;
+	float lpos_vx;
+	float lpos_vy;
+	float lpos_vz;
+	bool lpos_valid;
+	float lpos_evh;
+	float lpos_evv;
+	float att_q[4];
+	float air_pressure_pa;
+	float air_temperature_celsius;
+	float accel_z;
+	float vel_test_ratio;
+	float mag_test_ratio;
+	bool in_fixed_wing_flight;
+};
+
+class AirspeedValidator
+{
+public:
+	AirspeedValidator();
+	~AirspeedValidator();
+
+	float get_IAS() { return _IAS; }
+	float get_EAS() { return _EAS; }
+	float get_TAS() { return _TAS; }
+	bool get_airspeed_valid() { return _airspeed_valid; }
+	bool get_airspeed_failing() { return _airspeed_failing; }
+
+	bool get_data_stopped_fail() {return _data_stopped_failed;}
+	bool get_innov_check_fail() {return _innovations_check_failed;}
+	bool get_load_factor_check_fail() {return _load_factor_check_failed;}
+	float get_EAS_scale() {return _EAS_scale;}
+
+	void update_airspeed_validator(struct airspeed_validator_update_data &input_data);
+
+	void update_air_temperature(float temperature) {_air_temperature_celsius = temperature;}
+	void update_air_pressure(float pressure) {_air_pressure_pa = pressure;}
+
+	void update_wind_estimator_params();
+	void update_wind_estimator(uint64_t timestamp, float airspeed_true_raw, bool lpos_valid, float lpos_vx, float lpos_vy,
+				   float lpos_vz,
+				   float lpos_evh, float lpos_evv, float att_q[4]);
+	wind_estimate_s get_wind_estimator_states(uint64_t timestamp);
+	void update_EAS_scale();
+	void update_EAS_TAS(float air_pressure_pa, float air_temperature_celsius);
+	void check_airspeed_innovation(uint64_t timestamp, float estimator_status_vel_test_ratio,
+				       float estimator_status_mag_test_ratio);
+	void check_load_factor(float accel_z);
+	void update_airspeed_valid_status(uint64_t timestamp);
+
+	void set_wind_estimator_wind_p_noise(float wind_sigma) { _wind_estimator_wind_p_var = wind_sigma * wind_sigma; }
```

Comment:
@sfuhrer You need to be careful here. You are converting from a standard deviation to a variance. The wind estimator expects a standard deviation. So until now the wind estimator has been running with practically no process noise.